packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -206,6 +185,39 @@ export class Arn {
     });
   }
 
+  /**
+   * Extract the full resource name from an ARN
+   *
+   * Necessary for resource names (paths) that may contain the separator, like
+   * `arn:aws:iam::111111111111:role/path/to/role/name`.
+   *
+   * Only works if we statically know the expected `resourceType` beforehand, since we're going
+   * to use that to split the string on ':<resourceType>/' (and take the right-hand side).
+   *
+   * We can't extract the 'resourceType' from the ARN at hand, because CloudFormation Expressions
+   * only allow literals in the 'separator' argument to `{ Fn::Split }`, and so it can't be
+   * `{ Fn::Select: [5, { Fn::Split: [':', ARN] }}`.
+   *
+   * Only necessary for ARN formats for which the type-name separator is `/`.
+   */
+  public static extractResourceName(arn: string, resourceType: string): string {
```

Comment:
Keep the terminology consistent and name this `parseResourceName()`. It would help with discoverability as well.

Reply:
I will do what you ask in order to get this merged, but I'm against it. "parse resource name" is not a good name. The argument is not a resource name that needs parsing. The argument is an ARN that needs a part of it extracted.

Reply:
> I will do what you ask in order to get this merged

Don't do that please. 

You're right. `parseResourceName()` is not a good name. How about `parseResourceNameFromArn()`, `parseArnToResourceName()` or something else. I'm not good at names either.

If you still think `extract` is the best, I'm ok leaving it as is.

Reply:
I do think `extract` is better, so I guess I will change it back.

This PR is already taking a lot more effort than I was willing to put into it, given its priority among all the other things we should be doing :(.

Reply:
BTW,

> parseResourceNameFromArn

Given that the full invocation will already read `Arn.extractResourceName()`, I don't think there's much to be gained from `Arn.parseResourceNameFromArn()`.

Willing to consider something like `Arn.resourceNameFrom()` or something, but I don't really see the point in stuttering `Arn` a whole lot.

Reply:
`Arn.resourceNameFrom()` sounds nice